src/Integration.TeamExplorer/SectionController.cs

```diff
@@ -248,8 +244,8 @@ private void InitializeControllerCommands()
         {
             // Due to complexity of connect and bind we "outsource" the controlling part
             // to separate controllers which just expose commands
-            var connectionController = new Connection.ConnectionController(this.Host);
-            var bindingController = new Binding.BindingController(this.Host);
+            var connectionController = new Connection.ConnectionController(serviceProvider, Host);
+            var bindingController = new Binding.BindingController(serviceProvider, Host);
```

Comment:
TODO - we could probably now easily MEF-export the `ConnectionController` and `BindingController`, which would simplify the construction of the objects.